### ISSUE TITLE
`(include jbuild.inc)` does not copy `jbuild.inc` to the build directory

### ISSUE THREAD

**Issue by avsm** [ocaml/dune#431]
In RWO, we are using `(include jbuild.inc)` to generate some custom rules, and then a shell script rule calls `jbuilder` from one of them to evaluate some example files. 

I noticed that the target of the `(include )` directive is not copied into `_build` whereas the main `jbuild` file is.  It would be nice if `_build/` contained a complete copy of all the relevant `jbuild` files, so that a rule that recursively invokes jbuilder for something else within `_build` will work.

(note that the sub-jbuilds just generate their own `_build/` subdirectory for their invocations.  I hope that's not twisting custom rules too far :-)

**Comment by diml** [comment]
Jbuilder never copies files eagerly to the build directory, it only copy them as they are needed to perform the build. In particular, `jbuild` are special as they are currently read before starting the build, so they are read from the source tree and are never copied unless they are explicitly used as dependency of a rule.

I'm not sure I understand everything you are doing, but it seems to me that you might need a `(files_recursively_in ...)`.  BTW, would something like #429 do what you want?

**Comment by rgrinberg** [comment]
I'm closing this as this seems like a misunderstanding of how `include` works. @avsm it would be good if you'd describe what exactly you're using the OCaml syntax for btw. There's definitely some boilerplate required to define bulk rules/tests in dune.

**Comment by avsm** [comment]
We're not using this any more in rwo, but I'll follow up with a new issue if I run across this again.
